sdk/keyvault/azure_security_keyvault_secrets/README.md

```diff
@@ -1,24 +1,429 @@
+
+In order to interact with the Azure Key Vault service, you'll need to create an instance of the `SecretClient` struct. You need a **vault url**, which you may see as "DNS Name" in the portal, and credentials to instantiate a client object.
+
+The example shown below use a `DefaultAzureCredential`, which is appropriate for most scenarios including local development and production environments. Additionally, we recommend using a managed identity for authentication in production environments. You can find more information on different ways of authenticating and their corresponding credential types in the [Azure Identity] documentation.
+
+The `DefaultAzureCredential` will automatically pick up on an Azure CLI authentication. Ensure you are logged in with the Azure CLI:
+
+```azurecli
+az login
+```
+
+Instantiate a `DefaultAzureCredential` to pass to the client. The same instance of a token credential can be used with multiple clients if they will be authenticating with the same identity.
+
+### Set and Get a Secret
+
+```rust
+use azure_core::{RequestContent, Response};
+use azure_identity::DefaultAzureCredential;
+use azure_security_keyvault_secrets::{
+    models::{SecretBundle, SecretSetParameters},
+    SecretClient,
+};
+use serde_json::to_vec;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Create a new secret client
+    let credential = DefaultAzureCredential::new()?;
+    let client = SecretClient::new(
+        "https://your-key-vault-name.vault.azure.net/",
+        credential,
+        None,
+    )?;
+
+    // Create a new secret using the secret client.
+    let mut secret_set_parameters = SecretSetParameters::default();
+    secret_set_parameters.value = Some("secret-value".to_string());
+
+    // Serialize secret_set_parameters to Vec<u8>
+    let secret_set_parameters_bytes: Vec<u8> = to_vec(&secret_set_parameters)?;
+
+    let secret: Response<SecretBundle> = client
+        .set_secret(
+            "secret-name".to_string(),
+            RequestContent::from(secret_set_parameters_bytes),
+            None,
+        )
+        .await?;
+    println!("Response Code: {:?}", secret.status());
+
+    // Retrieve a secret using the secret client.
+    let secret: SecretBundle = client
+        .get_secret(
+            "secret-name".to_string(),
+            "secret-version".to_string(),
+            None,
+        )
+        .await?
+        .into_body()
+        .await?;
+    println!("{:?}", secret.value);
+
+    Ok(())
+}
+```
+
+## Key concepts
+
+### KeyVaultSecret
+
+A `KeyVaultSecret` is the fundamental resource within Azure Key Vault. From a developer's perspective, Azure Key Vault APIs accept and return secret values as strings.
+
+### SecretClient
+
+A `SecretClient` asynchronous operations in the SDK allowing for selection of a client based on an application's use case. Once you've initialized a `SecretClient`, you can interact with secrets in Azure Key Vault.
```

Comment:
"A" infers singular but "operations" is plural.
```suggestion
`SecretClient` asynchronous operations in the SDK allowing for selection of a client based on an application's use case. Once you've initialized a `SecretClient`, you can interact with secrets in Azure Key Vault.
```
This also doesn't make sense. It's the only client. To what selection do you refer? Did you start from Python or Java? Again, for KV anyway, I recommend basing it on .NET. I typically wrote the README first and shared it with other languages.

Reply:
I did take from .NET. I think the wording is weird because .NET has sync and async operations.